### ISSUE TITLE
jq -e decreases exit code by 10 when halt_error(number_bigger_then_9) is used

### ISSUE THREAD

**Issue by edek437** [jqlang/jq#2214]
**Describe the bug**
jq -e decreases exit code by 10 when halt_error(number_bigger_then_9) is used

**To Reproduce**
jq -en 'halt_error(9)'; echo $? # returns 9, it's ok
jq -en 'halt_error(10)'; echo $? # returns 0, 10 expected
jq -en 'halt_error(11)'; echo $? # returns 1, 11 expected
jq -en 'halt_error(110)'; echo $? # returns 100, 110 expected

**Environment (please complete the following information):**
 - OS and Version: 
Distributor ID: Ubuntu
Description:    Ubuntu 16.04.7 LTS
Release:        16.04
Codename:       xenial

 - jq version: 1.6

**Additional context**
NaN

**Comment by olorin37** [comment]
Or reproducing one-liner can be used:

```bash
for i in {0..255}; do jq -en "halt_error($i)"; echo "[$i] => $?"; done
```

Its output is located in file [output-jq.txt](https://github.com/stedolan/jq/files/5560123/output-jq.txt). It was also jq-1.6.

Distributor ID: openSUSE
Description:    openSUSE Tumbleweed
Release:        20201114

**Comment by itchyny** [comment]
This is a bug of 1.6, but resolved by #1697 (6d3d2750ec1e076cd86189b8fcfe1b760e2207c7) and already fixed in the master branch.
